compair/static/modules/answer/answer-form-partial.html

```diff
@@ -41,14 +41,17 @@ <h2><i class="fa fa-comments"></i> {{assignment.name}}</h2>
                 <p>Allowed file types for attachments include {{ UploadValidator.getAttachmentExtensionsForDisplay() }}, with a maximum file size of <strong>{{UploadValidator.getAttachmentUploadLimit() | filesize }}</strong>. Any files you upload will be attached when you <strong>save or submit this answer</strong> but not before.</p>
         </div>
 
-        <div class="form-group" ng-if="canManageAssignment && answer.draft && comparison_example !== true">
+        <div class="form-group" ng-if="canManageAssignment && answer.draft && comparison_example !== true" ng-class="{'has-error': answerForm.user_id.$invalid}">
             <label class="required-star" for="user_id">Submit As</label>
-            <select id="user_id" name="user_id" required
+            <select id="user_id" name="user_id" ng-required="validated"
                     ng-model="answer.user_id" class="form-control" chosen width="'100%'"
                     ng-options="u.id as u.fullname_sortable for u in classlist"
                     ng-change="onSubmitAsChanged(answer.user_id)">
                 <option value="">- Select a user -</option>
             </select>
+            <div class="form-control-error" ng-if="validated && answerForm.user_id.$invalid && answerForm.user_id.$error.required">
```

Comment:
We can try using ngMessages to handle the conditionals a bit cleaner https://code.angularjs.org/1.5.11/docs/api/ngMessages/directive/ngMessages . This should be especially helpful for handling multiple validation checks

so we could do something like
```
            <div ng-if="answerForm.$submitted" ng-messages="answerForm.user_id.$error">
                <div class="form-control-error" ng-message="required">Please select a user</div>
            </div>
```